lib/src/util/__tests__/customPropTypes.test.ts

```diff
@@ -43,4 +43,79 @@ describe('Custom PropTypes', function() {
       expect(retVal).toEqual(new TypeError(err));
     });
   });
+
+  describe('requiredPercentNumber', function() {
+    const componentName = 'Irrelevant',
+        propName = 'requiredPercent';
+
+    it('should return null if 0 is supplied', function() {
+      const props = { requiredPercent: 0 },
+          retVal = requiredPercentNumber(props, propName, componentName, '', '');
+
+      expect(retVal).toBeNull();
+    });
+
+    it('should return null if 100 is supplied', function() {
+      const props = { requiredPercent: 100 },
+          retVal = requiredPercentNumber(props, propName, componentName, '', '');
+
+      expect(retVal).toBeNull();
+    });
+
+    it('should return null if a number between 0 and 100 is supplied', function() {
+      const props = { requiredPercent: 50 },
+          retVal = requiredPercentNumber(props, propName, componentName, '', '');
+
+      expect(retVal).toBeNull();
+    });
+
+    it('should return an error if any other type of value is supplied', function() {
+      const props = { requiredPercent: '50' },
+          retVal = requiredPercentNumber(props, propName, componentName, '', ''),
+          err = `${componentName}: prop "${propName}" must be a number between 0 and 100 inclusive; received ` +
+              props[propName];
+
+      expect(retVal).toBeInstanceOf(TypeError);
+      expect(retVal).toEqual(new TypeError(err));
+    });
+
+    it('should return an error if a value less than 0 is supplied', function() {
+      const props = { requiredPercent: -1 },
+          retVal = requiredPercentNumber(props, propName, componentName, '', ''),
+          err = `${componentName}: prop "${propName}" must be a number between 0 and 100 inclusive; received ` +
+              props[propName];
+
+      expect(retVal).toBeInstanceOf(TypeError);
+      expect(retVal).toEqual(new TypeError(err));
+    });
+
+    it('should return an error if a value greater than 100 is supplied', function() {
+      const props = { requiredPercent: 101 },
+          retVal = requiredPercentNumber(props, propName, componentName, '', ''),
+          err = `${componentName}: prop "${propName}" must be a number between 0 and 100 inclusive; received ` +
+              props[propName];
+
+      expect(retVal).toBeInstanceOf(TypeError);
+      expect(retVal).toEqual(new TypeError(err));
+    });
+  });
```

Comment:
Should test when null and undefined are supplied to this one too

Reply:
> Should test when null and undefined are supplied to this one too

Done 0233f50e78105bae19a8d84cce928a4d31c13944

Reply:
> Done 0233f50

Do those tests pass?  In my reading of the component's source code, it doesn't seem to be written to error out in these cases but rather to clamp to the [0, 100] range.

Reply:
> Do those tests pass? In my reading of the component's source code, it doesn't seem to be written to error out in these cases but rather to clamp to the [0, 100] range.

Sorry, I forgot what these tests were for.  I was thinking of the component not the propTypes.  Nevermind.